libs/onboarding-sdk-product/README.md

```diff
@@ -60,4 +60,34 @@ Generally, you load product json string from an azure storage container, this ex
             throw new IllegalArgumentException("Found an issue when trying to serialize product json string!!");
         }
     }
+ ```
+
+## CacheableProductService
+
+This product has been implemented to work as a cacheable object, which keeps in memory the lastUpdated time of the
+products information.
+When the CacheableProduct gets created a DefaultProductService gets saved with its current state, and every time a
+method of the cacheable object is called, it will call a refresh method to check wether the product has been updated or
+not.
+If the product's last updatedTime is older than the retrieved current time from the azure client a new DefaultProduct
+will be saved into the cacheable Product.
```

Comment:
You could write something more readable....for example:

It is designed primarily focusing on maintaining the last updated `product.json` information in memory and the refresh is based on BlobProperties’s property **LastModified**.

When the **ProductServiceCacheable** gets instantiated a new `ProductServiceDefault` gets created reading the product.json file by azure storage and saving its current '**LastModified**' date.

At each method call, the service compares the product's lastUpdatedTime against the current time obtained from the Azure client. If the product's last updatedTime is older than the retrieved current time from the azure client a new `ProductServiceDefault` will be replaced the existing one.